tests/unit/processor/test_processor_strategy.py

```diff
@@ -89,3 +93,40 @@ def test_apply_processor_multiple_times_until_no_new_rule_matches(self):
             processor_metrics=mock_metrics,
         )
         assert expected_event == event
+
+    @pytest.mark.parametrize("execution_number", range(5))  # repeat test to ensure determinism
+    def test_strategy_applies_rules_in_deterministic_order(self, execution_number):
+        generic_tree = RuleTree()
+        specific_tree = RuleTree()
```

Comment:
the processor has already his trees

Reply:
I removed them, I did this also in the neighboring test and revised it a bit.